src/dune_lang/syntax.ml

```diff
@@ -43,6 +43,8 @@ module Version = struct
     parser_major = data_major && parser_minor >= data_minor
 end
 
+let inside_dune = ref false
```

Comment:
This doesn't look right to me. It doesn't impact only dune's own tests as other users sometimes also run dune inside dune. We could add a special version for such cases, for instance `(lang dune 2.1+dev)`. In this particular case, we might as well put the feature back in 2.0.0.